File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map <Character, Integer> check = new HashMap <> ();
        int max = 0;
        int l =0;
        for (int i =0; i<s.length();i++){
            if (check.get(s.charAt(i))==null){
                check.put(s.charAt(i),i);
            }
            else{
                l = Math.max(check.get(s.charAt(i)) + 1, l);
            }
            max=Math.max(max,1+i-l);
            check.put(s.charAt(i),i);
        }
        return max;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-8.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        check = {}
        second_check = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if s[i] not in check:
                check[s[i]] = 1
            elif s[i] in check:
                check[s[i]] = check[s[i]] + 1
            if t[i] in second_check:
                second_check[t[i]] = second_check[t[i]] + 1
            elif t[i] not in second_check:
                second_check[t[i]] = 1
        if check == second_check:
            return True
        print (check, second_check)
        return False
        